src/main/java/coupon/repository/DataSourceConfig.java

```diff
@@ -0,0 +1,56 @@
+package coupon.repository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.sql.DataSource;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.annotation.Primary;
+import org.springframework.jdbc.datasource.LazyConnectionDataSourceProxy;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+@Configuration
+public class DataSourceConfig {
+
+    @Bean
+    @ConfigurationProperties("coupon.datasource.writer")
+    public DataSource writeDataSource() {
+        return DataSourceBuilder.create().type(HikariDataSource.class).build();
+    }
+
+    @Bean
+    @ConfigurationProperties("coupon.datasource.reader")
+    public DataSource readDataSource() {
+        return DataSourceBuilder.create().type(HikariDataSource.class).build();
+    }
+
+    @Bean
+    @DependsOn({"writeDataSource", "readDataSource"})
+    public DataSource routeDataSource() {
+        DataSourceRouter dataSourceRouter = new DataSourceRouter();
+        DataSource writerDataSource = writeDataSource();
+        DataSource readerDataSource = readDataSource();
+
+        Map<Object, Object> dataSourceMap = new HashMap<>();
+        dataSourceMap.put(DataSourceRouter.WRITER_KEY, writerDataSource);
+        dataSourceMap.put(DataSourceRouter.READER_KEY, readerDataSource);
+
+        dataSourceRouter.setTargetDataSources(dataSourceMap);
```

Comment:
`writerDataSource` 선언 안 하고 이렇게 하는 건 어떨까요?
어차피 싱글톤이기도 하고 메서드 라인 줄일 수 있을 것 같아요!
취향에 따라 반영해주세요.

```suggestion
        dataSourceRouter.setTargetDataSources(Map.of(
                DataSourceRouter.WRITER_KEY, writeDataSource(),
                DataSourceRouter.READER_KEY, readDataSource()
        ));
```